### PR TITLE
tcontis/radio module refactor

### PR DESCRIPTION
**Pull request by tcontis**
## Description
A cleaner version of radio module refactor using RadioNode class.

## Associated Issue
Issue #1329

## Design Documents
[Link](link-to-design-doc)

### CODE REVIEW

File: soccer/Processor.cpp [hunk 1]
```diff
@@ -401,26 +395,7 @@ void Processor::run() {
...
-            unsigned int board = rx.robot_id();
-            if (board < Num_Shells) {
-                // We have to copy because the RX packet will survive past this
-                // frame but LogFrame will not (the RadioRx in LogFrame will be
-                // reused).
-                _context.state.self[board]->setRadioRx(rx);
-                _context.state.self[board]->radioRxUpdated();
-            }
-        }
+        if (_radio) curStatus.lastRadioRxTime = _radio->getLastRadioRxTime();
```
Comment (kylestach) [inline soccer/Processor.cpp:400]: I think eventually `curStatus` should be moved to `Context`, but that's not important yet.

File: soccer/Robot.hpp [hunk 2]
```diff
@@ -548,6 +548,7 @@ class OurRobot : public Robot {
     std::shared_ptr<Geometry2d::Circle> createBallObstacle() const;
 
     friend class Processor;
+    friend class RadioNode;
```
Comment (JNeiger) [inline soccer/Robot.hpp:551]: I don't think the `friend class Processor` is needed anymore

**Review by JNeiger — APPROVED** [review]
Everything looks pretty solid. Just get it passing style and it should be good to go! Nice job!

File: soccer/Robot.hpp (hunk 2 above)
Comment (tcontis) [reply soccer/Robot.hpp:551]: I believe it is, because Processor.cpp makes call to protected objects in the Robot class, such as angleFunctionPath.

**Comment by tcontis** [comment]
Code now working in simulation

File: soccer/Processor.hpp [hunk 3]
```diff
@@ -163,7 +165,7 @@ class Processor : public QThread {
 
     QMutex& loopMutex() { return _loopMutex; }
 
-    Radio* radio() { return _radio; }
+    Radio* radio() { return _radio->getRadio(); }
```
Comment (kylestach) [inline soccer/Processor.hpp:168]: Is this function used anywhere?

**Review by kylestach — APPROVED** [review]
Looks good other than my one question above.

File: soccer/Processor.hpp (hunk 3 above)
Comment (tcontis) [reply soccer/Processor.hpp:168]: In MainWindow.cpp:
https://github.com/RoboJackets/robocup-software/blob/cb819ca747de7c868b50851b678fee1bdc9ce5d8/soccer/ui/MainWindow.cpp#L1034-L1039

**Pull request merged**